kolibri/core/auth/models.py

```diff
@@ -159,6 +160,23 @@ class AbstractFacilityDataModel(FacilityDataSyncableModel):
     class Meta:
         abstract = True
 
+    def cache_related_dataset_lookup(self, related_obj_name):
+        """
+        Attempt to get the dataset_id either from the cache or the actual related obj instance.
+
+        :param related_obj_name: string representing the name of the related object on this model
+        :return: the dataset_id associated with the related obj
+        """
+        field = self._meta.get_field(related_obj_name)
+        key = '{id}_{db_table}_dataset'.format(id=getattr(self, field.attname), db_table=field.related_model._meta.db_table)
+        cached_value = cache.get(key)
```

Comment:
Mostly stylistic, but I quite like doing this as a single if statement:

```suggestion
        dataset_id = cache.get(key)
        if dataset_id is None:
            dataset_id = getattr(self, related_obj_name).dataset_id
            cache.set(key, dataset_id, 60 * 10)
        return dataset_id
```